conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -249,6 +255,11 @@ private MethodSpec createSetter(EnrichedField enriched) {
                 .addMember("value", "$S", enriched.fieldName().get());
         if (isCollectionType(type)) {
```

Comment:
isCollectionType applies to optional types as well as collections, I think we might need to take that into account below, and limit Optionals to applying nulls=SKIP